### PR TITLE
Add data source info manifest

### PR DESCRIPTION
**Pull request by keshav-725**

### CODE REVIEW

File: manifests/crc/default-db-included-installation/minikube/kruize-crc-minikube.yaml [hunk 1]
```diff
@@ -98,7 +98,28 @@ data:
         "port": 5432,
         "sslMode": "require",
         "username": "admin"
-      }
+      },
+      "datasource": [
```
Comment (msvinaykumar) [inline manifests/crc/default-db-included-installation/minikube/kruize-crc-minikube.yaml:102]: Recognize various methods of establishing a connection to a data source while implementing authentication. Furnish a list of at least three distinct approaches for connecting to a data source, detailing the necessary parameters. It's essential to adhere to compliance standards, which involve practices such as generating secure credentials to access confidential data. All these functionalities should be seamlessly supported using this JSON.

File: src/main/java/com/autotune/common/datasource/DataSourceInfo.java [hunk 2]
```diff
@@ -26,10 +26,14 @@
  * }
  */
 public class DataSourceInfo {
+    private final String name;
+    private final String serviceName;
     private final String provider;
     private final URL url;
 
-    public DataSourceInfo(String provider, URL url) {
+    public DataSourceInfo(String name, String serviceName, String provider, URL url) {
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/common/datasource/DataSourceInfo.java:34]: add more parametes like
HTTP Method:
Headers
Authentication Parameters:
Query Parameters:
 based o AUth type you may have to consider some more parameters or class

File: src/main/java/com/autotune/operator/InitializeDeployment.java [hunk 3]
```diff
@@ -157,5 +170,60 @@ private static String getKruizeConfigValue(String envName, JSONObject kruizeConf
             return envValue.toString();
     }
 
+    public static void populateDatasourceMapWithCongfig (HashMap<String, DataSourceInfo> datasourceMap) throws Exception {
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/operator/InitializeDeployment.java:173]: Generate an independent class for accessing data source information.

The provided code contains errors.

By executing kubectl apply -f above_crc.yaml, you will generate a configuration object or mount a configuration file. Subsequently, within your code, use either kubectl get config or read JSON data from the mounted file.

Develop a distinct class named "DataSource" within a separate package.

It's recommended to draft a design document prior to writing the code. Present and discuss this design document with  team. Following this, you can commence the coding process.

**Comment by dinogun** [comment]
Closing stale PRs

**Pull request closed without merging**